### PR TITLE
feat: split swap/limit/advanced orders tables

### LINKED ISSUE #2765: [1.42]Filter limit/advanced/market orders on pages
It would be nice to  separate orders displaying on the:
 - Swap page: market orders (Account modal)
 - Limit: show only limit orders
 - Advanced: show advanced orders only

[anxo] I'm fine either way for launch. But we should be coherent. If we want to filter for TWAP and only show TWAP in advance orders, we will need to filter also for limit orders.

Long term we might want to reiterate in twap, and aggregate parts into a single twap order. Once we do this, we might consider:
- Adding all types of order again together
- Considering the SWAP to have the orders being shown if they activate a setting

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #2765

Swap page: market orders (Account modal)
Limit orders page: show only limit orders
Advanced orders page: show advanced orders only

### CODE REVIEW

**Review by elena-zh — APPROVED** [review]
Great job!
I have created #2910 issue for the Advanced orders tab in order to change texts for Your orders table there.

File: src/common/hooks/useCategorizeRecentActivity.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export function useCategorizeRecentActivity() {
       allRecentActivity.reduce<[string[], string[]]>(
         (acc, activity) => {
           // Not order transactions (wrap, approve, etc.) doesn't have class property
-          if (!activity.class || activity.class !== OrderClass.LIMIT) {
+          if (!activity.class || activity.class === OrderClass.MARKET) {
```
Comment (anxolin) [inline src/common/hooks/useCategorizeRecentActivity.ts:22]: the comment above reads a bit strange, i get we want for recent history here MARKET orders and TX, but is expressed a bit strangely

File: src/pages/LimitOrders/index.tsx [hunk 2]
```diff
@@ -11,9 +16,15 @@ import {
...
 import * as styledEl from 'modules/trade/pure/TradePageLayout'
+import { useWalletInfo } from 'modules/wallet'
+
+import { getIsNotComposableCowOrder } from 'utils/orderUtils/getIsNotComposableCowOrder'
 
 export default function LimitOrderPage() {
+  const { chainId, account } = useWalletInfo()
   const { isUnlocked } = useAtomValue(limitOrdersRawStateAtom)
+  const allLimitOrders = useOrders(chainId, account, OrderClass.LIMIT)
+  const onlyPlainLimitOrders = useMemo(() => allLimitOrders.filter(getIsNotComposableCowOrder), [allLimitOrders])
```
Comment (anxolin) [inline src/pages/LimitOrders/index.tsx:27]: Are composable cows created with class LIMIT or TWAP?

**Review by anxolin — APPROVED** [review]

File: src/common/hooks/useCategorizeRecentActivity.ts (hunk 1 above)
Comment (alfetopito) [reply src/common/hooks/useCategorizeRecentActivity.ts:22]: I would rephrase it like:
// Only display regular onchain transactions (wrap, approval, etc) OR MARKET orders

**Review by alfetopito — COMMENTED** [review]
Since we are no longer showing MARKET orders in the limit orders table, we should go over the updaters and prevent them from being queried when not needed.
For example, filter them out of the spot price updater, do not try to calculate the estimated execution time (I think we already don't) and so on.

File: src/pages/LimitOrders/index.tsx (hunk 2 above)
Comment (shoom3301) [reply src/pages/LimitOrders/index.tsx:27]: Since `order-book` API returns all discrete orders, including TWAP parts we have to filter them.
The `order-book` API doesn't distinguish them, TWAP parts have `class=Limit`.
Because of it, we have to distinguish then on frontend side.

**Comment by shoom3301** [comment]
@alfetopito thanks! Added conditions to `UnfillableOrdersUpdater` and `SpotPricesUpdater`

File: src/legacy/state/orders/updaters/SpotPricesUpdater.ts [hunk 3]
```diff
@@ -25,26 +25,28 @@ type MarketRecord = Record<
...
   const pending = useCombinedPendingOrders({ chainId })
 
   return useSafeMemo(() => {
     return pending.reduce<Record<string, { chainId: number; inputCurrency: Token; outputCurrency: Token }>>(
       (acc, order) => {
-        if (chainId) {
-          // Aggregating pending orders per market. No need to query multiple times same market
-          const { marketInverted, marketKey } = getCanonicalMarketChainKey(chainId, order.sellToken, order.buyToken)
+        // Query spot prices only for Limit orders
+        if (order.class !== OrderClass.LIMIT) return acc
```
Comment (alfetopito) [inline src/legacy/state/orders/updaters/SpotPricesUpdater.ts:35]: We still show the market price and estimated market price on the TWAP orders table, no?

File: src/pages/LimitOrders/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/pages/LimitOrders/index.tsx:27]: Opened a thread here https://cowservices.slack.com/archives/C0375NV72SC/p1690199949260889
